src/components/cladeFilterDrawer/caseDefinitionConstructor/categoricalDataSelector.tsx

```diff
@@ -0,0 +1,54 @@
+import { useSelector, useDispatch } from "react-redux";
+import { caseDefFilter } from "../../../d";
+import Autocomplete from "@mui/material/Autocomplete";
+import TextField from "@mui/material/TextField";
+
+type CategoricalDataSelectorProps = {
+  field: string;
+};
+
+export const CategoricalDataSelector = (
+  props: CategoricalDataSelectorProps
+) => {
+  //@ts-ignore
+  const state = useSelector((state) => state.global);
+  const dispatch = useDispatch();
+  const { field } = props;
+  const metadataCensus = state.metadataCensus;
+
+  const allValues = metadataCensus[field]["uniqueValues"];
```

Comment:
Totally - I worry about this sort of thing. What's best practice here? Add a check / explicit error handling for each reference?

Reply:
It depends! If you were fully bought in to TypeScript, I think you'd want an explicit check. But it's also not crazy to have bugs break stuff. The bug wouldn't be with this component, it would be with the upstream setup, so having stuff break is not always terrible.

You're basically looking at various tradeoffs between safety (how badly does stuff break if this goes wrong), effort (how hard is it to handle it breaking / notice when it will break), value (how likely is this thing to break and thinking about the prior two), and visibility (how ugly will this look to the user if it breaks, combined with the prior three -- is it just visual? does the app require a restart/refresh? will we have data loss / bad results but the user won't notice something is wrong?).